src/Testcontainers.Pulsar/PulsarBuilder.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace Testcontainers.Pulsar;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class PulsarBuilder : ContainerBuilder<PulsarBuilder, PulsarContainer, PulsarConfiguration>
+{
+    public const string PulsarImage = "apachepulsar/pulsar:3.0.3";
+
+    public const ushort PulsarBrokerDataPort = 6650;
+
+    public const ushort PulsarWebServicePort = 8080;
+
+    public const string StartupScriptFilePath = "/testcontainers.sh";
+
+    public const string SecretKeyFilePath = "/pulsar/secret.key";
+
+    public const string Username = "test-user";
+
+    private static readonly IReadOnlyDictionary<string, string> AuthenticationEnvVars;
+
+    static PulsarBuilder()
+    {
+        const string authenticationPlugin = "org.apache.pulsar.client.impl.auth.AuthenticationToken";
+
+        var authenticationEnvVars = new Dictionary<string, string>();
+        authenticationEnvVars.Add("authenticateOriginalAuthData", "false");
+        authenticationEnvVars.Add("authenticationEnabled", "true");
+        authenticationEnvVars.Add("authorizationEnabled", "true");
+        authenticationEnvVars.Add("authenticationProviders", "org.apache.pulsar.broker.authentication.AuthenticationProviderToken");
+        authenticationEnvVars.Add("brokerClientAuthenticationPlugin", authenticationPlugin);
+        authenticationEnvVars.Add("CLIENT_PREFIX_authPlugin", authenticationPlugin);
+        authenticationEnvVars.Add("PULSAR_PREFIX_authenticationRefreshCheckSeconds", "5");
+        authenticationEnvVars.Add("PULSAR_PREFIX_tokenSecretKey", "file://" + SecretKeyFilePath);
+        authenticationEnvVars.Add("superUserRoles", Username);
+        AuthenticationEnvVars = new ReadOnlyDictionary<string, string>(authenticationEnvVars);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PulsarBuilder" /> class.
+    /// </summary>
+    public PulsarBuilder()
+        : this(new PulsarConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PulsarBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private PulsarBuilder(PulsarConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override PulsarConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <returns></returns>
+    public PulsarBuilder WithAuthentication()
+    {
+        return Merge(DockerResourceConfiguration, new PulsarConfiguration(authenticationEnabled: true))
+            .WithEnvironment(AuthenticationEnvVars);
+    }
```

Comment:
Do we really need this API? It looks like Java does not support it either, and it prevents us from using the same [wait strategy](https://github.com/testcontainers/testcontainers-java/blob/13c2641ff1ff8e74f37594bff96275811a7d809b/modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java#L109) Java uses. I have serious issues with the current wait strategy. Running `bin/pulsar-admin clusters list` seems not reliable. Tests are flaky locally.

Reply:
@HofmeisterAn In our company's workflow, we heavily depend on WithAuthentication, especially during testing. Bringing this functionality into TestContainers would greatly help facilitate the adoption of TestContainers to all our engineers. While examining the Java version's withStrategy, I encountered some challenges in implementing the wait strategy for token authentication. Nonetheless, I'm determined to explore potential solutions and see if I can successfully integrate it

Reply:
If it helps, we can split the PR into two separate PRs. One that adds the module configuration, similar to Java, and a follow-up that specifically adds the authentication support.